### PR TITLE
Move some helpers for pathnavigation to Ghoul

### PR DESCRIPTION
**Pull request by WeirdRubberDuck**
interpolation, numerical integration and quaternions

### CODE REVIEW

File: include/ghoul/misc/interpolator.h [hunk 1]
```diff
@@ -58,6 +76,31 @@ T interpolateLinear(double t, const T& p0, const T& p1);
...
 T interpolateCatmullRom(double t, const T& p0, const T& p1, const T& p2, const T& p3);
 
+/**
+ * Returns an interpolated value using a Catmull-Rom spline interpolation scheme based
+ * on an implementation by Mika Rantanen:
+ * https://qroph.github.io/2018/07/30/smooth-paths-using-catmull-rom-splines.html
+ *
+ * Four control points are used in the interpolation; \p p0, \p p1, \p p2, and \p p3.
+ * The interpolation is calculated based on the interpolation factor \p t and the
+ * parameter \p alpha, that controls the shape of the curve. A smaller alpha leds to
```
Comment (WeirdRubberDuck) [inline include/ghoul/misc/interpolator.h:86]: typo for "alpha" here and "resulting" below

File: include/ghoul/glm.h [hunk 2]
```diff
@@ -117,6 +117,25 @@ constexpr bool isGlmVector() {
         glm::uvec2, glm::uvec3, glm::uvec4>::value;
 }
 
+/*
+ * Compute a quaternion that represents the rotation looking from \p eye to
+ * \p target, with the specified \p up direction
+ */
+template <typename valType>
+glm::tquat<valType> lookAtQuaternion(glm::tvec3<valType> eye, glm::tvec3<valType> target,
```
Comment (alexanderbock) [inline include/ghoul/glm.h:125]: `const glm::tvec3<valType>` instead?

File: include/ghoul/glm.h [hunk 3]
```diff
@@ -117,6 +117,25 @@ constexpr bool isGlmVector() {
         glm::uvec2, glm::uvec3, glm::uvec4>::value;
 }
 
+/*
```
Comment (alexanderbock) [inline include/ghoul/glm.h:120]: `/**` instead to make it Doxygen compatible

File: include/ghoul/misc/integration.inl [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+        { -0.5384693, 0.47862867 },
+        { 0.5384693, 0.47862867 },
+        { -0.90617985, 0.23692688 },
+        { 0.90617985, 0.23692688 }
+    };
+
+    const double a = t0;
+    const double b = t1;
+    T sum = 0.0;
+    for (auto coefficient : coefficients) {
```
Comment (alexanderbock) [inline include/ghoul/misc/integration.inl:74]: `const GaussLegendreCoefficient&` instead of `auto`?

File: src/misc/interpolator.cpp [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+glm::dquat interpolateLinear(double t, const glm::dquat& p0, const glm::dquat& p1) {
+    return glm::slerp(p0, p1, t);
+}
+
+// Based on implementation by Mika Rantanen, but without tension
+// https://qroph.github.io/2018/07/30/smooth-paths-using-catmull-rom-splines.html
+glm::dvec3 interpolateCatmullRom(double t, const glm::dvec3& p0, const glm::dvec3& p1,
+                                 const glm::dvec3& p2, const glm::dvec3& p3, double alpha)
+{
+    ghoul_assert(t >= 0 && t <= 1.0, "Interpolation variable must be in range [0,1]");
```
Comment (alexanderbock) [inline src/misc/interpolator.cpp:47]: `0.0` instead of `0`

File: src/misc/interpolator.cpp [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+    ghoul_assert(alpha >= 0 && alpha <= 1.0, "Alpha must be in range [0,1]");
```
Comment (alexanderbock) [inline src/misc/interpolator.cpp:48]: Same here

**Review by alexanderbock — APPROVED** [review]

**Pull request merged**